txmongo/connection.py

```diff
@@ -241,7 +241,7 @@ class ConnectionPool(object):
     __wc_possible_options = set(['w', "wtimeout", 'j', "fsync"])
 
     def __init__(self, uri="mongodb://127.0.0.1:27017", pool_size=1, ssl_context_factory=None, **kwargs):
-        assert isinstance(uri, str)
+        assert isinstance(uri, StringType)
```

Comment:
I would rather get rid of all asserts and just raise errors. We can do that in another refactor branch?